tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -220,3 +220,79 @@ ZTEST_F(ascs_test_suite, test_abort_client_operation_if_callback_not_registered)
 		      param->response_code);
 	zassert_equal(0x00, param->reason, "unexpected Reason 0x%02x", param->reason);
 }
+
+ZTEST_F(ascs_test_suite, test_release_ase_on_acl_disconnection_client_terminates_cis)
+{
+	const struct bt_gatt_attr *ase;
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	struct bt_iso_chan *chan;
+	uint8_t ase_id;
+
+	if (IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK)) {
+		ase = fixture->ase_snk.attr;
+		ase_id = fixture->ase_snk.id;
+	} else {
+		ase = fixture->ase_src.attr;
+		ase_id = fixture->ase_src.id;
+	}
+
+	zexpect_not_null(ase);
+	zexpect_true(ase_id != 0x00);
+
+	bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
+
+	/* Set ASE to non-idle state */
+	test_preamble_state_streaming(conn, ase_id, stream, &chan,
+				      !IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK));
+
+	/* Mock ACL disconnection */
+	mock_bt_conn_disconnected(conn);
+
+	/* Mock CIS disconnection */
+	mock_bt_iso_disconnect(chan);
+
+	/* Expected to notify the upper layers */
+	expect_bt_bap_unicast_server_cb_release_called_once(stream);
+	expect_bt_bap_stream_ops_released_called_once(stream);
+
+	bt_bap_unicast_server_unregister_cb(&mock_bap_unicast_server_cb);
+}
+
+ZTEST_F(ascs_test_suite, test_release_ase_on_acl_disconnection_server_terminates_cis)
+{
+	const struct bt_gatt_attr *ase;
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	struct bt_iso_chan *chan;
+	uint8_t ase_id;
+
+	if (IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK)) {
+		ase = fixture->ase_snk.attr;
+		ase_id = fixture->ase_snk.id;
+	} else {
+		ase = fixture->ase_src.attr;
+		ase_id = fixture->ase_src.id;
+	}
+
+	zexpect_not_null(ase);
+	zexpect_true(ase_id != 0x00);
+
+	bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
+
+	/* Set ASE to non-idle state */
+	test_preamble_state_streaming(conn, ase_id, stream, &chan,
+				      !IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK));
+
+	/* Mock ACL disconnection */
+	mock_bt_conn_disconnected(conn);
+
+	/* Server waits for CIS to be disconnected by client */
+	k_sleep(K_MSEC(CONFIG_BT_ASCS_ISO_DISCONNECT_DELAY));
+
+	/* Expected to notify the upper layers */
+	expect_bt_bap_unicast_server_cb_release_called_once(stream);
+	expect_bt_bap_stream_ops_released_called_once(stream);
```

Comment:
The `CONFIG_BT_ASCS_ISO_DISCONNECT_DELAY` is only used to disconnect the CIS if the stream goes out of the streaming state; it is not related to CIS disconnect events due to ACL disconnects. 

I don't see the value of this specific test case, as neither the client nor the server initiates the CIS disconnect if the ACL disconnects

Reply:
Cannot agree, as ACL disconnection triggers the `ascs_disconnect_stream` if CIS has been not disconnected before ACL (which is possible case - allowed by the spec).

Reply:
Hmm, I see your point, however there is not really any value in attempting to disconnect the CIS _after_ the ACL disconnect event: We know the CIS will get disconnected right after (if not before, depending on the controller implementation). 

IMO the host should not attempt to disconnect the CIS on ACL disconnect, which I can see that we do in ASCS - Maybe that part needs changed? There no value in calling `ascs_disconnect_stream` if ACL state is not `BT_CONN_STATE_CONNECTED`.